crates/uv-distribution-types/src/prioritized_distribution.rs

```diff
@@ -603,3 +634,50 @@ impl IncompatibleWheel {
         }
     }
 }
+
+/// Given a wheel filename, determine the set of supported platforms, in terms of their markers.
+pub fn implied_markers(filename: &WheelFilename) -> MarkerTree {
+    let mut marker = MarkerTree::FALSE;
+    for platform_tag in &filename.platform_tag {
+        match platform_tag.as_str() {
+            "any" => marker.or(MarkerTree::TRUE),
+            tag if tag.starts_with("win") => {
+                marker.or(MarkerTree::expression(MarkerExpression::String {
+                    key: MarkerValueString::SysPlatform,
+                    operator: MarkerOperator::Equal,
+                    value: "win32".to_string(),
+                }));
+            }
+            tag if tag.starts_with("manylinux") => {
+                marker.or(MarkerTree::expression(MarkerExpression::String {
+                    key: MarkerValueString::SysPlatform,
+                    operator: MarkerOperator::Equal,
+                    value: "linux".to_string(),
+                }));
+            }
+            tag if tag.starts_with("musllinux") => {
+                marker.or(MarkerTree::expression(MarkerExpression::String {
+                    key: MarkerValueString::SysPlatform,
+                    operator: MarkerOperator::Equal,
+                    value: "linux".to_string(),
+                }));
+            }
+            tag if tag.starts_with("linux") => {
+                marker.or(MarkerTree::expression(MarkerExpression::String {
+                    key: MarkerValueString::SysPlatform,
+                    operator: MarkerOperator::Equal,
+                    value: "linux".to_string(),
+                }));
+            }
+            tag if tag.starts_with("macosx") => {
+                marker.or(MarkerTree::expression(MarkerExpression::String {
+                    key: MarkerValueString::SysPlatform,
+                    operator: MarkerOperator::Equal,
+                    value: "darwin".to_string(),
+                }));
+            }
+            _ => {}
+        }
+    }
+    marker
```

Comment:
We basically have control over how granular we want to make this (and the forking logic). For example... we could easily include architecture here too, so that we ensure we always get an x86 and an ARM wheel for macOS. However, we'd then need to _require_ (in the resolver) that we always find an x86 wheel somewhere.

So the tradeoff here is: we can get more complete resolutions (making it less likely that we fail at install time due to a missing wheel), but we make the resolver stricter (making it more likely that resolution fails entirely).

Reply:
Another way to think of this: we can make sure that there's _always_ an x86 wheel available for macOS, but that requires that we can always _find_ an x86 wheel for macOS.

Reply:
It would be ideal if we could make those very specific, that is include the arch, but give the user control over what they need. For example, if the user only want macos (any), than both the x86_64 and the aarch64 wheel should satisfy that since they are more specific than macos.

Reply:
I don’t think I understand. Is this different then what’s in the follow-up PR?

Reply:
Yes this is covered by #10017

Reply:
Ah right, makes sense.